File: 01-grokking-the-coding-interview/01-sliding-window/examples/01_max_sum_of_contigous_subarray_of_size_n/java/MainApp/src/main/java/io/github/mainapp/MainApp.java
package io.github.mainapp;

public class MainApp {
    /**
    * Find the max sum of subarray of a fixed size 3
    * Example input:
    * [4, 2, 1, 7, 8, 1, 2, 8, 1, 0]
    * === 16
    */
    public int Run(int [] arr, int k) {
        int maxValue = Integer.MIN_VALUE;
        int currentRunningSum = 0;

        for (int i = 0; i < arr.length; i++)
        {
            currentRunningSum += arr[i];
            if (i >= k - 1)
            {
                maxValue = Math.max(maxValue, currentRunningSum);
                currentRunningSum -= arr[i - (k - 1)];
            }
        }
        return maxValue;
    }
}

File: 01-grokking-the-coding-interview/01-sliding-window/examples/01_max_sum_of_contigous_subarray_of_size_n/python/MainApp/app/mainApp.py
import sys


class MainApp:
    def __init__(self):
        pass
    '''
    Find the max sum of subarray of a fixed size 3
    Example input:
    [4, 2, 1, 7, 8, 1, 2, 8, 1, 0]
    === 16
    '''
    def run(self, arr, k):
        max_value = -sys.maxsize - 1
        current_running_sum = 0

        for idx, ele in enumerate(arr):
            current_running_sum += arr[idx]

            if idx >= k - 1:
                max_value = max(current_running_sum, max_value)
                current_running_sum -= arr[idx - (k - 1)]
        return max_value
